libsplinter/src/admin/service/shared.rs

```diff
@@ -705,6 +789,66 @@ impl AdminServiceShared {
                 self.current_consensus_verifiers = verifiers;
                 Ok((expected_hash, proto_circuit_proposal))
             }
+            CircuitManagementPayload_Action::CIRCUIT_DISBAND_REQUEST => {
+                debug!("Circuit disband request being processed");
+                let circuit_id = circuit_payload
+                    .get_circuit_disband_request()
+                    .get_circuit_id();
+
+                // Validate the specified circuit exists
+                self.admin_store.get_circuit(circuit_id)
+                    .map_err(|err| {
+                        AdminSharedError::ValidationFailed(format!(
+                            "error occurred when trying to get circuit {}",
+                            err
+                        ))
+                    })?
+                    .ok_or_else(|| {
+                        AdminSharedError::ValidationFailed(format!(
+                            "Received disband request for a circuit that does not exist: circuit id {}",
+                            circuit_id
+                        ))
+                    })?;
+
+                // Creating the proposal to disband this circuit
+                let circuit_proposal = self.make_disband_request_circuit_proposal(
+                    circuit_id,
+                    header.get_requester(),
+                    header.get_requester_node_id(),
+                )?;
+                let mut verifiers = vec![];
+                let mut protocol = ADMIN_SERVICE_PROTOCOL_VERSION;
+                for member in circuit_proposal.get_circuit_proposal().get_members() {
+                    verifiers.push(admin_service_id(member.get_node_id()));
+                    // Figure out what protocol version should be used for this proposal
+                    if let Some(protocol_version) = self
+                        .service_protocols
+                        .get(&admin_service_id(member.get_node_id()))
+                    {
+                        if protocol_version < &protocol {
+                            protocol = *protocol_version
```

Comment:
This protocol isnt used. If the protocol version is 1, it should reject this proposal right away since Disband is not supported.

Reply:
Fixed to send this protocol to the `validate_disband_circuit` method, and added a check to this method for the protocol (returning an error if it's 1) and added a test `test_validate_disband_circuit_invalid_protocol` to validate this.